Content.Server/GameObjects/Components/Weapon/Melee/StunbatonComponent.cs

```diff
@@ -63,31 +99,175 @@ public override bool OnHitEntities(IReadOnlyList<IEntity> entities)
                     stunnable.Slowdown(_slowdownTime);
             }
 
+            cell.DeductCharge(Wattage);
+            if(cell.Charge < Wattage)
+            {
+                _entitySystemManager.GetEntitySystem<AudioSystem>()
+                    .Play(AudioHelpers.GetRandomFileFromSoundCollection("sparks"), Owner.Transform.GridPosition, AudioHelpers.WithVariation(0.25f));
+                TurnOff();
+            }
+
             return false;
         }
 
-        public bool UseEntity(UseEntityEventArgs eventArgs)
+        private bool ToggleStatus(IEntity user)
         {
+            if (Activated)
+            {
+                TurnOff();
+            }
+            else
+            {
+                TurnOn(user);
+            }
+
+            return true;
+        }
+
+        private void TurnOff()
+        {
+            if (!_activated)
+            {
+                return;
+            }
+
             var sprite = Owner.GetComponent<SpriteComponent>();
             var item = Owner.GetComponent<ItemComponent>();
 
+            SoundComponent soundComponent;
+
+            if (Owner.TryGetComponent(out soundComponent))
+            {
+                soundComponent.Play("/Audio/machines/button.ogg");
+            }
```

Comment:
Please replace all the instances of SoundComponent with AudioSystem instead. SoundComponent isn't meant for playing one-off sounds like these.

Reply:
You got it chief